server/control.go

```diff
@@ -311,23 +312,43 @@ func (c Controller) runOnce(ctx context.Context, leaderKey string, tick <-chan t
 		return err
 	}
 
-	re, err := inf.Storage().GetRebootsEntries(ctx)
+	rqEntries, err := inf.Storage().GetRebootsEntries(ctx)
 	if err != nil {
 		return err
 	}
-	metrics.UpdateReboot(len(re))
+	metrics.UpdateRebootQueueEntries(len(rqEntries))
+	rqEntries = cke.DedupRebootQueueEntries(rqEntries)
+	itemCounts := cke.CountRebootQueueEntries(rqEntries)
+	metrics.UpdateRebootQueueItems(itemCounts)
 
-	var reboot *cke.RebootQueueEntry
-	if len(re) > 0 {
+	if len(rqEntries) > 0 {
 		disabled, err := inf.Storage().IsRebootQueueDisabled(ctx)
 		if err != nil {
 			return err
 		}
-		if !disabled {
-			reboot = re[0]
+		if disabled {
+			rqEntries = nil
 		}
 	}
-	ops, phase := DecideOps(cluster, status, constraints, rcs, reboot)
+
+	nf := NewNodeFilter(cluster, status)
+	apiServers := map[string]bool{}
+	for _, node := range cluster.Nodes {
+		if nf.nodeStatus(node).APIServer.Running {
```

Comment:
I'm not sure whether we should check the running API server or the expected API server, i.e. `nf.ControlPlane()`.

Reply:
In this case, "expected API servers" should be used.

`apiServers` slice is used to check whether any of API servers are rebooting. If an API server is rebooting, "running API servers" might not contain the server. It will cause miss-detection.

So, I will fix the line.